lib/vector-core/src/enrichment/mod.rs

```diff
@@ -0,0 +1,16 @@
+pub mod tables;
+
+pub use tables::{TableSearch, Tables};
+pub use vrl_core::Condition;
+
+/// Enrichment tables represent additional data sources that can be used to enrich the event data
+/// passing through Vector.
+pub trait Table: std::fmt::Debug {
+    /// Search the enrichment table data with the given condition.
+    /// All fields within the data must match (AND).
+    fn find_table_row(&self, condition: Vec<Condition>) -> Option<&Vec<String>>;
```

Comment:
There's a lot of owned things being passed here (i.e. `Vec` itself and the two `String`s within `Condition`) and it doesn't seem ideal to have to reallocate them for every call. Could we just pass a reference instead?

Reply:
Most of the Condition needs to be evaluated per event and thus needs allocating. I can borrow the key as that is fairly static.

I've made this change in the next [PR](https://github.com/timberio/vector/blob/1bfc06f3e75609b5124f50c5d7b6b09d57cb7c45/lib/vrl/compiler/src/enrichment_tables.rs#L8-L11).